components/formstack/formstack.app.mjs

```diff
@@ -0,0 +1,105 @@
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "app",
+  app: "formstack",
+  propDefinitions: {
+    formId: {
+      type: "string",
+      label: "Forms",
+      description: "Select a form",
+      async options({ page }) {
+        const forms = await this.getForms({
+          page: page + 1,
+        });
+
+        return forms.map((form) => ({
+          label: form.name,
+          value: form.id,
+        }));
+      },
+    },
+  },
+  methods: {
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return "https://www.formstack.com/api/v2";
+    },
+    async _makeRequest(path, options = {}, $ = this) {
+      return axios($, {
+        ...options,
+        url: `${this._apiUrl()}/${path}`,
+        headers: {
+          ...options.headers,
+          "Authorization": `Bearer ${this._accessToken()}`,
+          "Content-Type": "application/json",
+        },
+      });
+    },
+    async createWebhook({
+      formId, data,
+    }) {
+      return this._makeRequest(`form/${formId}/webhook.json`, {
+        method: "post",
+        data: {
+          ...data,
+          content_type: "json",
+          handshake_key: this._accessToken(),
+        },
+      });
+    },
+    async removeWebhook(webhookId) {
+      return this._makeRequest(`webhooks/${webhookId}.json`, {
+        method: "delete",
+      });
+    },
+    async getForms({ $ }) {
```

Comment:
I would recommend creating the function `getForms` by just implementing what the description suggests in this case
```javascript
     async getForms(args = {}) {
       const {
         $,
         params,
       } = args;
       const options = {
         params,
       };
       return this._makeRequest("form", options, $);
     },
```
and then create a generic function to paginate over resources depending on the implemented function eg.
```javascript
    async paginateResources({
      resourceFn,
      resourceFnArgs = {},
      mapOnResponse = (response) => response.forms?.map((form) => form),
      maxPages = 10,
    }) {
      let page = resourceFnArgs.params?.page || 1;
      let allResources = [];

      do {
        const response =
          await resourceFn({
            ...resourceFnArgs,
            params: {
              page,
              per_page: 100,
              ...resourceFnArgs.params,
            },
          });

        const nextResources = mapOnResponse(response);

        page = nextResources.length === 100
          ? page + 1
          : 0;

        allResources = allResources.concat(nextResources);

      } while (page > 0 && page <= maxPages);

      return allResources;
    },
```
then you can paginate any resource like this
```javascript
    const forms = await this.paginateResources({
        resourceFn: this.getForms,
        resourceArgs: {
          params: {
            page: 1,
            per_page: 100,
          },
        },
        mapOnResponse: (response) => response.forms?.map((form) => form),
        maxPages: 10,
      });
```
Of course you can improve this strategy, but the idea is to be as generic as possible with code so we can reuse it later on.